### PR TITLE
Add development setup document and modify config files as needed

### PR DESCRIPTION
**Pull request by kasabe28**
# Proposed Changes

- Add development setup document
- Modify config files as needed

### CODE REVIEW

File: config/apinetlet/default/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -24,3 +24,4 @@ spec:
         - "--health-probe-bind-address=:8081"
         - "--metrics-bind-address=127.0.0.1:8080"
         - "--leader-elect"
+        - "--api-net-namespace=apinetlet-system"
```
Comment (balpert89) [inline config/apinetlet/default/manager_auth_proxy_patch.yaml:27]: This is a potential problem, the net resources should be handled in `ironcore-net-system`.

```suggestion
        - "--api-net-namespace=ironcore-net-system"
```

Comment (afritzler) [reply config/apinetlet/default/manager_auth_proxy_patch.yaml:27]: This flag actually should not be part of the patch as the `namespace` is typically patched as n arg on the deployment size. Default behavior should be not setting this argument.

File: config/apinetlet/default/kustomization.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ namePrefix: apinetlet-
 resources:
 - ../rbac
 - ../manager
+- ../apinet-rbac
```
Comment (balpert89) [inline config/apinetlet/default/kustomization.yaml:18]: @afritzler I am not sure if this should be employed here - i can remember this should only be done when all components run together in the same cluster.

If this is the case, the inclusion of `./apinet-rbac` should be done downstream as well depending of this, e.g. in the kind-specific kustomization configuration:

```
resources:
  - github.com/ironcore-dev/ironcore-net/config/apinetlet/apinet-rbac?ref=437081e6eea333713fa75ce6ccf3a46025a8dac0
  - github.com/ironcore-dev/ironcore-net/config/apinetlet/default?ref=437081e6eea333713fa75ce6ccf3a46025a8dac0
```

The same goes for `metalnetlet-rbac`

**Review by afritzler — CHANGES_REQUESTED** [review]

File: docs/development/setup.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+### Setup `ironcore-net` with any other kubernetes cluster
+
+For local development with any kubernetes cluster other than `kind`, a make target that applies the manifests using 
+provided images is available via
+
+1. Install ironcore-net API server and API services into the K8s cluster.
+
+```shell
+make install APISERVER_IMG=<apiserver image>
```
[the remark this answers is not in the document]
Comment (kasabe28) [reply docs/development/setup.md]: removed steps for clusters other than kind

File: config/apinetlet/default/kustomization.yaml (hunk 2 above)
Comment (kasabe28) [reply config/apinetlet/default/kustomization.yaml]: Addressed

**Review by afritzler — APPROVED** [review]

**Pull request merged**